Rules/CommonScripts/PlayerNames.as

```diff
@@ -0,0 +1,129 @@
+#define CLIENT_ONLY
+
+float spacing = 1.0f; // spacing between hearts
+Vec2f icon_size = Vec2f(12, 12); // heart icon frame size
+
+float max_radius = 64.0f; // screenspace distance
+uint16[] blob_ids;
+
+void onTick(CRules@ this)
+{
+	if (!this.canShowHoverNames() || g_videorecording)
+		return;
+	
+	blob_ids.clear();
+	CControls@ c = getControls();
+	Vec2f mouse_pos = c.getMouseWorldPos();
+
+	uint8 team = this.getSpectatorTeamNum();
+	CBlob@ my_blob = getLocalPlayerBlob();
+	if (my_blob !is null) // dont change team if we are dead, so that we can see everyone names
+		team = my_blob.getTeamNum();
+
+	for (int i = 0; i < getPlayerCount(); i++)
+	{
+		CPlayer@ player = getPlayer(i);
```

Comment:
because of an engine bug, player can be null here if someone gets kicked from the server- add a null check